kafka-cron/producer/main_test.go

```diff
@@ -0,0 +1,165 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/Shopify/sarama"
+	"github.com/Shopify/sarama/mocks"
+	"github.com/berkeli/kafka-cron/types"
+	"github.com/berkeli/kafka-cron/utils"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_PublishMessages(t *testing.T) {
+	prod := mocks.NewSyncProducer(t, nil)
+
+	clusters := []string{"cluster-a", "cluster-b"}
+	wantMsg := "hello world"
+	prefix := "jobs"
+	topicPrefix = &prefix
+
+	prod.ExpectSendMessageWithMessageCheckerFunctionAndSucceed(func(msg *sarama.ProducerMessage) error {
+		bytes, err := msg.Value.Encode()
+		if err != nil {
+			return err
+		}
+
+		gotMsg := string(bytes)
+
+		if gotMsg != wantMsg {
+			return fmt.Errorf("got message %s, want %s", gotMsg, wantMsg)
+		}
+
+		if msg.Topic != "jobs-cluster-a" {
+			return fmt.Errorf("got topic %s, want %s", msg.Topic, "jobs-cluster-a")
+		}
+
+		return nil
+	})
+
+	prod.ExpectSendMessageWithMessageCheckerFunctionAndSucceed(func(msg *sarama.ProducerMessage) error {
+		bytes, err := msg.Value.Encode()
+		if err != nil {
+			return err
+		}
+
+		gotMsg := string(bytes)
+
+		if gotMsg != wantMsg {
+			return fmt.Errorf("got message %s, want %s", gotMsg, wantMsg)
+		}
+
+		if msg.Topic != "jobs-cluster-b" {
+			return fmt.Errorf("got topic %s, want %s", msg.Topic, "jobs-cluster-b")
+		}
+
+		return nil
+	})
+
+	PublishMessages(prod, wantMsg, clusters)
+}
+
+func Test_CommandPublisher(t *testing.T) {
```

Comment:
Do you have any tests around the actual repetition aspect of the schedule parsing? i.e. any tests which show that commands will get run more than once?